api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java

```diff
@@ -431,7 +431,13 @@ public DbWorkspace setPublished(
   @Override
   public void publishWorkspaceViaDB(
       String workspaceNamespace, PublishWorkspaceRequest publishWorkspaceRequest) {
-    final DbWorkspace dbWorkspace = workspaceDao.getByNamespace(workspaceNamespace).orElseThrow();
+    final DbWorkspace dbWorkspace =
+        workspaceDao
+            .getByNamespace(workspaceNamespace)
+            .orElseThrow(
+                () ->
+                    new NotFoundException(
```

Comment:
this exception subclasses `WorkbenchException`, which means it gets auto-magically translated to a 404 error code at the controller level.